### PR TITLE
feat: expose wal config

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- Allow users to config wal parameters thru config files.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/datanode/src/datanode.rs [hunk 1]
```diff
@@ -47,6 +47,35 @@ impl Default for ObjectStoreConfig {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WalConfig {
+    // wal directory
+    pub dir: String,
+    // wal file size in bytes
+    pub file_size: usize,
```
Comment (evenyag) [inline src/datanode/src/datanode.rs:55]: Could we use crates like [bytesize](https://github.com/hyunsik/bytesize) to serialize/deserialize the size?
```suggestion
    pub file_size: ByteSize,
```

File: src/datanode/src/datanode.rs [hunk 2]
```diff
@@ -47,6 +47,35 @@ impl Default for ObjectStoreConfig {
...
+    // wal purge threshold in bytes
+    pub purge_threshold: usize,
+    // purge interval in seconds
+    pub purge_interval: u64,
```
Comment (evenyag) [inline src/datanode/src/datanode.rs:59]: We have imported the `humantime-serde` crate, we could use it.
https://github.com/GreptimeTeam/greptimedb/blob/72f05a3137bdedf0e7502a246a2533ded8d104cc/src/servers/src/http.rs#L111-L112

File: src/datanode/src/datanode.rs (hunk 1 above)
Comment (v0y4g3r) [reply src/datanode/src/datanode.rs:55]: Looks like the `serde` feature is still under development: https://github.com/hyunsik/bytesize/pull/29

**Review by waynexia — APPROVED** [review]

File: src/datanode/src/datanode.rs (hunk 1 above)
Comment (v0y4g3r) [reply src/datanode/src/datanode.rs]: I copy `ReadableSize` from [raft-engine](https://github.com/tikv/raft-engine/blob/8dd2a39f359ff16f5295f35343f626e0c10132fa/src/util.rs#L23) to make size config items human-readable.

**Pull request merged**